src/libsystemd-network/sd-radv.c

```diff
@@ -987,3 +1062,76 @@ int sd_radv_route_prefix_set_lifetime(sd_radv_route_prefix *p, uint64_t lifetime
 
         return 0;
 }
+
+int sd_radv_pref64_prefix_new(sd_radv_pref64_prefix **ret) {
+        sd_radv_pref64_prefix *p;
+
+        assert_return(ret, -EINVAL);
+
+        p = new(sd_radv_pref64_prefix, 1);
+        if (!p)
+                return -ENOMEM;
+
+        *p = (sd_radv_pref64_prefix) {
+                .n_ref = 1,
+
+                .opt.type = RADV_OPT_PREF64,
+                .opt.length = 2,
+                .prefixlen = 96,
+                .lifetime_usec = RADV_DEFAULT_VALID_LIFETIME_USEC,
+        };
+
+        *ret = p;
+        return 0;
+}
+
+DEFINE_PUBLIC_TRIVIAL_REF_UNREF_FUNC(sd_radv_pref64_prefix, sd_radv_pref64_prefix, mfree);
+
+int sd_radv_pref64_prefix_set_prefix(
+                sd_radv_pref64_prefix *p,
+                const struct in6_addr *in6_addr,
+                unsigned char prefixlen,
+                uint32_t valid_lifetime) {
+
+        uint8_t prefixlen_code = 0;
+
+        assert_return(p, -EINVAL);
+        assert_return(in6_addr, -EINVAL);
+
+        if (prefixlen > 128)
+                return -EINVAL;
+
+        if (prefixlen > 96)
+                log_radv(NULL, "Unusual PREF64 prefix length %u greater than 64", prefixlen);
```

Comment:
```suggestion
        if (prefixlen > 96)
                log_radv(NULL, "Unusual PREF64 prefix length %u greater than 96", prefixlen);
```

Condition and message mismatch, the condition is correct, message is off.